Sources_v3/Controllers/ChannelController/ChannelController.swift

```diff
@@ -695,12 +750,22 @@ public extension _ChatChannelController where ExtraData == DefaultExtraData {
     }
 }
 
+/// Describes the flow of the items in the list
+public enum ListOrdering {
+    /// New items appear on the top of the list.
+    case topToBottom
+    
+    /// New items appear on the bottom of the list.
+    case bottomToTop
+}
+
 // MARK: - Delegates
 
-/// `ChannelController` uses this protocol to communicate changes to its delegate.
+/// `ChatChannelController` uses this protocol to communicate changes to its delegate.
 ///
 /// This protocol can be used only when no custom extra data are specified. If you're using custom extra data types,
-/// please use `ChannelControllerDelegateGeneric` instead.
+/// please use `_ChatChannelControllerDelegate` instead.
```

Comment:
create your own **typealias ChatChannelControllerDelegate**?

Reply:
Oh, I wish it was possible to do that! Unfortunately, it's not possible to do that in Swift yet. See https://github.com/apple/swift/blob/master/docs/GenericsManifesto.md#generalized-existentials